client/app/components/dashboards/dashboard-widget/VisualizationWidget.jsx

```diff
@@ -27,8 +27,12 @@ function visualizationWidgetMenuOptions({ widget, canEditDashboard, onParameters
   const canEditParameters = canEditDashboard && !isEmpty(invoke(widget, "query.getParametersDefs"));
   const widgetQueryResult = widget.getQueryResult();
   const isQueryResultEmpty = !widgetQueryResult || !widgetQueryResult.isEmpty || widgetQueryResult.isEmpty();
-
-  const downloadLink = fileType => widgetQueryResult.getLink(widget.getQuery().id, fileType);
+  const parts = window.location.pathname.split('/').reverse()
```

Comment:
These codes are unmaintainable and may not work in some browsers. I have mentioned this code before and it was rejected. If you want to do this function, I suggest rewriting a public page component to support it. It's my mind.

And i don't know `parts[0].length === 40` meaning? i think it's unmaintaineable.

Reply:
Just my personal opinion.